File: Year02/2nd semester/MPP/Lab/mpp-proiect-csharp-AlexandruPaul21/CsharpProj/model/Booking.cs
namespace CsharpProj.model;

public class Booking : Entity<long>
{
    public Flight Flight { get; set; }
    public Client Client { get; set; }
    public List<string> Passengers { get; set; }

    public Booking(Flight flight, Client client, List<string> passengers)
    {
        Flight = flight;
        Client = client;
        this.Passengers = passengers;
    }
}

File: Year02/2nd semester/MPP/Lab/mpp-proiect-java-AlexandruPaul21/JavaProjectNetworking/ProjectModel/src/main/java/org/example/Booking.java
package org.example;

import java.util.List;

public class Booking extends Entity<Long> {

    private Flight flight;
    private Client client;
    private List<String> passengers;

    public Booking(Flight flight, Client client, List<String> passengers) {
        this.flight = flight;
        this.client = client;
        this.passengers = passengers;
    }

    public Flight getFlight() {
        return flight;
    }

    public void setFlight(Flight flight) {
        this.flight = flight;
    }

    public Client getClient() {
        return client;
    }

    public void setClient(Client client) {
        this.client = client;
    }

    public List<String> getPassengers() {
        return passengers;
    }

    public void setPassengers(List<String> passengers) {
        this.passengers = passengers;
    }

    @Override
    public String toString() {
        return "Booking{" +
                "flight=" + flight +
                ", client=" + client +
                ", passengers=" + passengers +
                '}';
    }
}
